### DISCUSSION TITLE
Anvil as inspiration

### DISCUSSION THREAD

**Discussion by nathvi** [elm-time/elm-time#14] [Ideas]
I've been experimenting with [Anvil](https://anvil.works/) for a while, and at a high level, it offers a full stack web development experience using "nothing but python".

Might be worth looking at the free version, and seeing what they offer. If there was something like Anvil, that used Elm / Haskell, I would be ecstatic. 

Pure functional programming + strict typing is the only comfortable way to build larger, more maintainable web apps. 

I know this project is aimed more at helping students not wanting to deal with setup. Anvil has found a market of both students and professional programmers, who ALSO don't want to be bogged down by the soul-deadening mindless work of setting up and maintaining dev environments.

I'm thinking about the difference between racecar drivers and the pit crew. I just want to drive. I don't want to change tires.

**Comment by Viir** [comment]
Thank you for the pointer. I was not aware of Anvil.

My impression so far is that Anvil is at a much later stage of building a commercial product. So to take inspiration from Anvil specifically, we'd probably look into the past: How did they get the resources to develop to today's stage? What did they offer two years ago?

The evolution of the project seems to be headed in a similar direction as you described. Elm Fullstack also supports professional applications already. The challenges of users and customers mainly drive the development so far. Ideas for increasing the speed or taking shortcuts are welcome.👍
